src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1011,25 +1010,41 @@ private void BlockDownloaded(uint256 blockHash, Block block, int peerId)
                     throw new InvalidOperationException("Unsolicited block");
                 }
 
-                if (block != null)
+                chainedHeader = this.chainedHeaderTree.GetChainedHeader(blockHash);
+                if (chainedHeader == null)
                 {
-                    chainedHeader = this.chainedHeaderTree.GetChainedHeader(blockHash);
+                    lock (this.blockRequestedLock)
+                    {
+                        this.callbacksByBlocksRequestedHash.Remove(blockHash);
+                    }
+
+                    this.logger.LogTrace("(-)[CHAINED_HEADER_NOT_FOUND]");
+                    return;
+                }
 
-                    if (chainedHeader == null)
+                if (block == null)
```

Comment:
code starting from this line doesn't have to be in peer lock

when it's outside of the peer lock there is no need for `bool reassignDownload`, reassigning can be done right away

Reply:
@dangershony suggested that it needs to be in a lock as `chainedHeader.BlockDataAvailability` can change whilst we check the block availability state?

Reply:
If I am not mistaken any access to the ChainedHeader statuses should happen inside CHT and under a lock (in this case we do it outside CHT) we could move this check in to CHT (similar to other places where we call CHT only to change a status)

Reply:
>@dangershony suggested that it needs to be in a lock as chainedHeader.BlockDataAvailability can change whilst we check the block availability state?

Fair point, it's possible. 

Then please ignore my comment.